src/gz/bufread.rs

```diff
@@ -639,3 +731,156 @@ impl<R: AsyncWrite + BufRead> AsyncWrite for MultiGzDecoder<R> {
         self.get_mut().shutdown()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::gz::bufread::*;
+    use std::io;
+    use std::io::{Cursor, Read, Write};
+
+    //a cursor turning EOF into blocking errors
+    #[derive(Debug)]
+    pub struct BlockingCursor {
```

Comment:
I do not know if there is such a structure ready in rust implementing `Read` and where we can specify at some points that we received more data